src/components/Checkbox/Checkbox.tsx

```diff
@@ -33,42 +45,63 @@ export class Checkbox extends React.Component<ICheckboxProps, {}> implements ICh
     } = this.props;
 
     return (
-      <div className={ css('ms-ChoiceField', className) }>
+      <div
+        className={ css('ms-Checkbox', className) }
+        ref={ (c): HTMLElement => this._checkBox = c }
+      >
         <input
           { ...inputProps }
           { ...(checked !== undefined && { checked }) }
           { ...(defaultChecked !== undefined && { defaultChecked }) }
           disabled={ disabled }
-          ref={ (el): HTMLInputElement => this._checkBox = el }
+          ref={ (el): HTMLInputElement => this._checkBoxInput = el }
           id={ this._id }
           name={ this._id }
-          className='ms-ChoiceField-input'
+          className='ms-Checkbox-input'
           type='checkbox'
-          role='checkbox'
           onChange={ this._onChange }
           aria-checked={ checked }
         />
-        <label htmlFor={ this._id } className='ms-ChoiceField-field'>
+        <label htmlFor={ this._id }
+          ref={ (el): HTMLLabelElement => this._checkBoxLabel = el }
+          className={ css('ms-Checkbox-label', (checked || defaultChecked) && 'is-checked', disabled && 'is-disabled') }
+        >
           { label && <span className='ms-Label'>{ label }</span> }
         </label>
       </div>
     );
   }
 
   public get checked(): boolean {
-    return this._checkBox ? this._checkBox.checked : false;
+    return this._checkBoxInput ? this._checkBoxInput.checked : false;
   }
 
   public focus() {
-      if (this._checkBox) {
-          this._checkBox.focus();
+      if (this._checkBoxInput) {
+          this._checkBoxInput.focus();
       }
   }
 
+  private _onFocus(): void {
+    this._checkBox.classList.add('is-inFocus');
```

Comment:
if you use React eventing, which you should, you may want to make sure you're not stomping on something passed in via props.